File: client/hrms-client/src/services/JobSeekerService.js
import axios from "axios"

const url = "http://localhost:4000/api/jobsekkers/"


const getAll = async ()=>{
      const res = await axios.get(url+"getall")

      return res.data
}

const getById = async (id)=>{
      const res = await axios.get(url+"getById?id="+id)

      return res.data
}

const register = async (job_seeker)=>{
      const res = await axios.post(url+"register")

      return res.data
}

const login = async (email,password)=>{
      const res = await axios.post(url+"login",{email:email,password:password})

      return res.data
}

const getExperiencesByUserId = async (id)=>{
      const res = await axios.get(url+"getExperiencesByUserId?id="+id)

      return res.data
}

const getSchoolsByUserId = async (id) =>{
      const res = await axios.get(url+"getSchoolsByUserId?id="+id)

      return res.data
}

const getForeignLanguagesByUserId = async (id)=>{
      const res = await axios.get(url+"getForeignLanguagesByUserId?id="+id)

      return res.data
}

const getSkillsByUserId = async (id)=>{
      const res = await axios.get(url+"getSkillsByUserId?id="+id)

      return res.data
}

const uploadImage = async (id,imgFile)=>{
      const res = await axios.post(url+"uploadimage",{id,imgFile})

      return res.data
}

const service = {getAll,getById,register,login,getExperiencesByUserId,getSchoolsByUserId,getForeignLanguagesByUserId,getSkillsByUserId,uploadImage}


export default service

File: server/hrms/src/main/java/kodlamaio/hrms/business/abstracts/JobSeekerService.java
package kodlamaio.hrms.business.abstracts;

import kodlamaio.hrms.core.utilities.results.Result;
import kodlamaio.hrms.entities.concretes.*;
import kodlamaio.hrms.entities.dtos.EmailDto;
import kodlamaio.hrms.entities.dtos.JobSeekerRegisterDto;

import java.util.List;

public interface JobSeekerService {
    List<JobSeeker> getall();

    Result getById(int id);

    Result register(JobSeekerRegisterDto jobSeekerRegisterDto);
    Result MailConfirm(EmailDto emailDto);

    Result add(JobSeeker jobSeeker);

    Result uploadImage(int id,String imgUrl);

    List<School> getSchoolsByUserId(int id);

    List<Experience> getExperiencesByUserId(int id);

    List<ForeignLanguage> getForeignLanguagesByUserId(int id);

    List<Skill> getSkillsByUserId(int id);

    Result login(String eposta,String password);
}
